packages/amplify-category-function/src/provider-utils/awscloudformation/utils/getDependentFunction.ts

```diff
@@ -27,10 +27,12 @@ export async function lambdasWithApiDependency(
     const selectedCategories = currentParameters.permissions;
     let deletedModelFound: boolean;
 
-    for (const selectedResources of Object.values(selectedCategories)) {
-      deletedModelFound = Object.keys(selectedResources).some(r => modelsDeleted.includes(r));
-      if (deletedModelFound) {
-        dependentFunctions.push(lambda);
+    if (selectedCategories !== undefined && selectedCategories.length !== 0) {
```

Comment:
Is `selectedCategories` supposed to be an object? If so, you can use `typeof selectedCategories === 'object' && selectedCategories !== null`.

Reply:
Ping @akshbhu

Reply:
Done!